doc/manual/manual.tex

```diff
@@ -2268,10 +2268,87 @@ \subsection{Particles}
+
+\item Compile \& Install (advanced users and developers with the need to reconfigure underlying libraries or running massively parallel models): The most advanced option is to compile and install \aspect{} from source. This allows maximal control over the underlying libraries like Trilinos and deal.II, as well as easy modifications to \aspect{} by recompiling a modified source directory. Our installation instructions cover most Linux and MacOS operating systems, but incompatibilities on individual systems can always occur and make the installation more cumbersome. If you are planning to run massively parallel computations on a compute cluster this is likely your only option. Since clusters usually have a very individual setup, it is always a good advice to ask IT support staff for help when installing \aspect{}, to avoid hard to reproduce setup problems, and performance penalties.
+\end{enumerate}
+
+\subsection{Docker Container}
+
+\subsubsection{Installing Docker and downloading the \aspect{} image}
+
+Docker is a lightweight virtualization software that allows to ship applications with all their dependencies in a simple way. It is outside of the scope of this manual to explain all possible applications of Docker, and we refer to the introduction (\url{https://www.docker.com/what-docker}) and installation and quickstart guides (\url{https://www.docker.com/products/docker}) on the Docker website for more detailed descriptions of how to set up and use the docker engine. More importantly Docker provides a marketplace for exchanging prepared docker images (called Docker Hub). After setting up the docker engine downloading a precompiled \aspect{} image from Docker Hub is as simple as typing in a terminal:
+
+\begin{verbatim}
+docker pull gassmoeller/aspect
+\end{verbatim}
+
+\subsubsection{Running \aspect{} models}
+Although it is possible to use the downloaded \aspect{} docker image in a number of different ways, we recommend the following workflow:
+
+\begin{enumerate}
+\item Create your \aspect{} input file in a folder of your choice (possibly also containing any input model data that is required by your model).
+\item Run the docker image and mount the prepared directory as a volume into the docker container. The code below assumes you have navigated into the prepared directory. When you have started the container run the aspect model inside the container, and make sure the model output is saved in your prepared directory.
+\begin{verbatim}
+docker run -it -v "$(pwd):/home/dealii/aspect/model" \
```

Comment:
Looks good to me, the ideal container depends on the use case.  So far I've come up with:
1) runtime container - small as possible, good for tutorials, filing bug reports, ruling out that a problem isn't specific to a custom build, and potentially even production runs.
2) full source code/build from scratch container - good for CI testing, helping developers get started, allowing users to tune parameters, etc.  Sneaky sysadmins might even peak at the build scripts to clarify exactly how a release is built.

In either case it's pretty straight forward to just write what a user expects to their home directory.  That way input and output files could be kept, even if the user ditches the containers and upgrades to newer versions or builds from scratch without the containers.

How to handle dependencies is an interesting question.  Sure in an ideal world Deal.II would have a source code container that built a release into a runtime container.  Then the Aspect-devel container would use the deal.II runtime to build into a aspect-runtime container. 

For the file/permission issues I've use tinkered with entrypoint.sh and passing in the UID, that way the container can write to my homedir with my user ID.  My testing without it results in mounting my directory inside a docker container and then having new files owned by root when created.